iface/tun_windows.go

```diff
@@ -32,6 +36,40 @@ func (c *tunDevice) Create() error {
 	return c.assignAddr()
 }
 
+// createWithUserspace Creates a new Wireguard interface, using wireguard-go userspace implementation
+func (c *tunDevice) createWithUserspace() (NetInterface, error) {
+	tunIface, err := tun.CreateTUN(c.name, c.mtu)
+	if err != nil {
+		return nil, err
+	}
+
+	// We need to create a wireguard-go device and listen to configuration requests
+	tunDevice := device.NewDevice(tunIface, c.iceBind, device.NewLogger(device.LogLevelSilent, "[wiretrustee] "))
+	err = tunDevice.Up()
+	if err != nil {
+		return tunIface, err
+	}
+
+	uapi, err := c.getUAPI(c.name)
```

Comment:
I do not see where do you close the uapi resource

Reply:
Just added this to the Close() method